plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -398,6 +400,42 @@ instance PlutusTx.Eq ScriptPurpose where
     a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
   _ == _ = Haskell.False
 
+-- | Like `ScriptPurpose` but also contains redeemers, and an optional datum
+-- for spending scripts.
+data ScriptInfo
+  = MintingScript V2.CurrencySymbol V2.Redeemer
```

Comment:
It wasn't clear to me whether we were sure we wanted the redeemer in every constructor like this or as a separate product field.

Reply:
@zliu41 Sorry to bother you, but I am working on integrating this into ledger and I remember we both agreed that it'll be a good idea to move `Redeemer` into `ScriptContext` instead of putting into every constructor of `ScriptInfo`

Whenever you get a chance could you make this change, if we are on the same page about this that is?

Reply:
Yes I'm going to do that